vendor/src/github.com/docker/libnetwork/networkdb/networkdb.go

```diff
@@ -223,9 +223,10 @@ func (nDB *NetworkDB) getEntry(tname, nid, key string) (*entry, error) {
 // table, key) tuple and if the NetworkDB is part of the cluster
 // propogates this event to the cluster. It is an error to create an
 // entry for the same tuple for which there is already an existing
-// entry.
+// entry unless the current entry is deleting state.
 func (nDB *NetworkDB) CreateEntry(tname, nid, key string, value []byte) error {
-	if _, err := nDB.GetEntry(tname, nid, key); err == nil {
+	e, _ := nDB.getEntry(tname, nid, key)
```

Comment:
func (nDB *NetworkDB) GetEntry(tname, nid, key string) ([]byte, error) {

i feel the e is not  in correct position.